docs/lang/articles/reference.md

```diff
@@ -0,0 +1,172 @@
+---
+sidebar_position: 9998
+---
+
+# Language Reference
+
+This article describes the syntax and semantics of the Taichi programming
+language.
+
+**To users**: If you have gone through user tutorials and still feel uncertain
+about your program behavior, then you are coming to the correct place. If you
+find the actual behavior different from the one described in this article, feel
+free to create an [issue](https://github.com/taichi-dev/taichi/issues/new/choose).
+Anything unspecified in this article is subject to change, so you should not
+rely on it in your programs.
+
+**To contributors**: This article specifies what the language *should* be. That
+is, you should try to match the implementation of the Taichi compiler with this
+article. You can clearly determine a certain behavior is *correct*, *buggy*, or
+*undefined* from this article.
+
+## Introduction
+
+Taichi is a domain-specific language embedded in Python.
+[Kernels and functions](basic/syntax.md) clearly defines the boundary between
+the Taichi language and the Python language - code in the Taichi scope is
+treated as the former, while code in the Python scope is treated as the latter.
+It should be emphasized that this article is about *the Taichi language*.
+
+That said, because Taichi is embedded in Python, the syntax of Taichi is a
+subset of that of Python. To make life easier, this article is modeled after
+the [Python language reference](https://docs.python.org/3/reference/). The
+[notation](https://docs.python.org/3/reference/introduction.html#notation) and
+[lexical analysis](https://docs.python.org/3/reference/lexical_analysis.html)
+parts exactly follow Python, so familiarize yourself with them if you are new
+to them.
+
+## Basic concepts
+
+Before detailing syntax and semantics in the next few chapters, many important
```

Comment:
many basic but important concepts and general evaluation principles specific to Taichi are listed here.